py/desiutil/annotate.py

```diff
@@ -0,0 +1,376 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+# -*- coding: utf-8 -*-
+"""
+=================
+desiutil.annotate
+=================
+
+Tools for adding units and comments to FITS files.
+"""
+import csv
+import os
+import sys
+from argparse import ArgumentParser
+import yaml
+from astropy.io import fits
+from astropy.table import Table, QTable
+from astropy.units import UnitConversionError
+from . import __version__ as desiutilVersion
+from .log import get_logger, DEBUG
+
+
+log = None
+
+
+def csv_unit_column(header, comment=False):
+    """Find the column that contains unit descriptions, or comments.
+
+    Parameters
+    ----------
+    header : iterable
+        The column names from a CSV file.
+    comment : :class:`bool`, optional
+        If ``True``, look for a column matching 'comment'.
+
+    Returns
+    -------
+    :class:`int`
+        The index of `header` that matches.
+
+    Raises
+    ------
+    IndexError
+        If no match was found.
+    """
+    search = ('unit', )
+    if comment:
+        search = ('comment', 'description')
+    for i, column in enumerate(header):
+        for s in search:
+            if s in column.lower():
+                return i
+    raise IndexError(f"No column matching '{search[0]}' found!")
+
+
+def csv_units(filename):
```

Comment:
Minor: let's prefix this with a verb like `read_csv_units` or `load_csv_units`.

Reply:
Will do.

Reply:
Done in [dab51a4](https://github.com/desihub/desiutil/pull/199/commits/dab51a474c40e19cc3dbbc9c2b8cfd0d4d4e199b).